### ISSUE TITLE
Fails to build guile for target i686-w64-mingw32.static due to indentation warning and -Werror being specified

### ISSUE THREAD

**Issue by jakint0sh** [mxe/mxe#3055]
Attempting to build package `guile` for target `i686-w64-mingw32.static` fails due to an indentation warning, and `-Werror` being specified in the compiler options.  The error reads:
```
../../guile-1.8.8/libguile/gc-segment.c: In function 'scm_i_find_heap_segment_containing_object':
../../guile-1.8.8/libguile/gc-segment.c:413:3: error: this 'if' clause does not guard... [-Werror=misleading-indentation]
  413 |   if ((scm_t_cell* ) obj < lowest_cell || (scm_t_cell*) obj >= highest_cell)
      |   ^~
../../guile-1.8.8/libguile/gc-segment.c:417:5: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'if'
  417 |     {
      |     ^
cc1: all warnings being treated as errors
make[5]: *** [Makefile:1037: libguile_la-gc-segment.lo] Error 1
```
Full terminal output and logfile: https://gist.github.com/jakint0sh/7c1bc3475ba711d4e30b601be927e2d6

I suspect this is because older versions of gcc did not warn you about indentation errors like this.  I suggest adding `-Wno-misleading-indentation` to the compiler options for this package.

**Comment by mabrand** [comment]
Can you make a pull request for this?

**Comment by jakint0sh** [comment]
Sure, but gimme a few days because I need to go learn how this whole project works \:P  I can troubleshoot compiler errors, but I really have no idea of what cross-compilation and patching wizardry MXE does, and I have less of an idea of how it goes about doing it.

**Comment by jakint0sh** [comment]
Okay!  Just figured out how to fix this and a pull request is open: #3062
